pkg/peertls/tlsopts/options_test.go

```diff
@@ -104,34 +109,39 @@ func TestNewOptions(t *testing.T) {
 	}
 }
 
-func pregeneratedIdentity(t *testing.T) *identity.FullIdentity {
-	const chain = `-----BEGIN CERTIFICATE-----
-MIIBQDCB56ADAgECAhB+u3d03qyW/ROgwy/ZsPccMAoGCCqGSM49BAMCMAAwIhgP
-MDAwMTAxMDEwMDAwMDBaGA8wMDAxMDEwMTAwMDAwMFowADBZMBMGByqGSM49AgEG
-CCqGSM49AwEHA0IABIZrEPV/ExEkF0qUF0fJ3qSeGt5oFUX231v02NSUywcQ/Ve0
-v3nHbmcJdjWBis2AkfL25mYDVC25jLl4tylMKumjPzA9MA4GA1UdDwEB/wQEAwIF
-oDAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwDAYDVR0TAQH/BAIwADAK
-BggqhkjOPQQDAgNIADBFAiEA2ZvsR0ncw4mHRIg2Isavd+XVEoMo/etXQRAkDy9n
-wyoCIDykUsqjshc9kCrXOvPSN8GuO2bNoLu5C7K1GlE/HI2X
------END CERTIFICATE-----
------BEGIN CERTIFICATE-----
-MIIBODCB4KADAgECAhAOcvhKe5TWT44LqFfgA1f8MAoGCCqGSM49BAMCMAAwIhgP
-MDAwMTAxMDEwMDAwMDBaGA8wMDAxMDEwMTAwMDAwMFowADBZMBMGByqGSM49AgEG
-CCqGSM49AwEHA0IABIZrEPV/ExEkF0qUF0fJ3qSeGt5oFUX231v02NSUywcQ/Ve0
-v3nHbmcJdjWBis2AkfL25mYDVC25jLl4tylMKumjODA2MA4GA1UdDwEB/wQEAwIC
-BDATBgNVHSUEDDAKBggrBgEFBQcDATAPBgNVHRMBAf8EBTADAQH/MAoGCCqGSM49
-BAMCA0cAMEQCIGAZfPT1qvlnkTacojTtP20ZWf6XbnSztJHIKlUw6AE+AiB5Vcjj
-awRaC5l1KBPGqiKB0coVXDwhW+K70l326MPUcg==
------END CERTIFICATE-----`
-
-	const key = `-----BEGIN EC PRIVATE KEY-----
-MHcCAQEEIKGjEetrxKrzl+AL1E5LXke+1ElyAdjAmr88/1Kx09+doAoGCCqGSM49
-AwEHoUQDQgAEoLy/0hs5deTXZunRumsMkiHpF0g8wAc58aXANmr7Mxx9tzoIYFnx
-0YN4VDKdCtUJa29yA6TIz1MiIDUAcB5YCA==
------END EC PRIVATE KEY-----`
-
-	fi, err := identity.FullIdentityFromPEM([]byte(chain), []byte(key))
-	assert.NoError(t, err)
+func TestPeerCAWhitelist(t *testing.T) {
+	t.Run("all nodes signed", func(t *testing.T) {
+		testplanet.Run(t, testplanet.Config{
+			SatelliteCount:   1,
+			StorageNodeCount: 10,
+			UplinkCount:      0,
+			UsePeerCAWhitelist: true,
+		}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+			time.Sleep(5 * time.Second)
+
+			err := planet.Ping(ctx)
+			assert.NoError(t, err)
+		})
+	})
+
+	t.Run("unsigned satellite", func(t *testing.T) {
+		testIdentities, err := testplanet.MixedIdentities([]int{1}, testplanet.MixedIndexesUnsigned)
```

Comment:
any reason you are not using here `testplanet.PregeneratedUnsigned(0)` for that particular node. I think this would avoid the need of having `MixedIdentities` func.

Reply:
can you elaborate?

Reply:
This relies on the ability to reconfigure a particular identity for a node.